fendermint/vm/interpreter/src/chain.rs

```diff
@@ -56,28 +84,63 @@ where
         state: Self::State,
         msg: Self::Message,
     ) -> anyhow::Result<(Self::State, Self::DeliverOutput)> {
+        let (pool, state) = state;
+
         match msg {
             ChainMessage::Signed(msg) => {
                 let (state, ret) = self
                     .inner
                     .deliver(state, VerifiableMessage::Signed(msg))
                     .await?;
-                Ok((state, ChainMessageApplyRet::Signed(ret)))
-            }
-            ChainMessage::Ipc(_) => {
-                // This only happens if a validator is malicious or we have made a programming error.
-                // I expect for now that we don't run with untrusted validators, so it's okay to quit.
-                todo!("#191: implement execution handling for IPC")
+                Ok(((pool, state), ChainMessageApplyRet::Signed(ret)))
             }
+            ChainMessage::Ipc(msg) => match msg {
+                IpcMessage::BottomUpResolve(msg) => {
+                    let smsg = relayed_bottom_up_ckpt_to_fvm(&msg)
+                        .context("failed to syntesize FVM message")?;
+
+                    // Let the FVM validate the checkpoint quorum certificate and take not of the relayer for rewards.
```

Comment:
Let me see if I understand this. The idea is that: 
- There is a `Resolve` message committing the checkpoint which includes the `Cid` of the list of bottom-up messages to be executed. 
- If the commitment happens, the `Cid` of the messages are queue for resolution and a posterior proposal an execution?

I still think we may need to remove the resolution stage and directly include messages as part of the checkpoint in a first implementation. Message resolution may include additional complexity in the L2->root interaction :(  

In any case, I feel this will make more sense as I move towards the implementation of the actual execution.

Reply:
Yes, you got that right, and as per our sync we will implement both behaviours depending on what kind of parent we'll be dealing with. For Fendermint<-Fendermint relationships we will use this, for Lotus<-Fendermint we won't.